index.html

```diff
@@ -204,55 +204,1951 @@ <h3>
+              presenting them with the list of devices in |availableDevices|.
+              </li>
+              <li>If the user does not choose a device, [=queue a global task=]
+              on the [=relevant global object=] of [=this=] using the [=HID
+              device task source=] to [=reject=] |promise| with an
+              {{"AbortError"}} {{DOMException}} and abort these steps.
+              </li>
+              <li>Let |device:HIDDevice| be the {{HIDDevice}} instance chosen
+              by the user.
+              </li>
+              <li>[=Queue a global task=] on the [=relevant global object=] of
+              [=this=] using the [=HID device task source=] to [=resolve=]
+              |promise| with |device|.
+              </li>
+            </ol>
+          </li>
+          <li>Return |promise|.
+          </li>
+        </ol>
+        <section data-dfn-for="HIDDeviceRequestOptions">
+          <h4>
+            <dfn>HIDDeviceRequestOptions</dfn> dictionary
+          </h4>
+          <pre class="idl">
+            dictionary HIDDeviceRequestOptions {
+                required sequence&lt;HIDDeviceFilter&gt; filters;
+            };
+          </pre>
+          <dl>
+            <dt>
+              <dfn>filters</dfn> member
+            </dt>
+            <dd>
+              Filters for HID devices.
+            </dd>
+          </dl>
+        </section>
+        <section data-dfn-for="HIDDeviceFilter">
+          <h4>
+            <dfn>HIDDeviceFilter</dfn> dictionary
+          </h4>
+          <pre class="idl">
+            dictionary HIDDeviceFilter {
+                unsigned long vendorId;
+                unsigned short productId;
+                unsigned short usagePage;
+                unsigned short usage;
+            };
+          </pre>
+          <dl>
+            <dt>
+              <dfn>vendorId</dfn> member
+            </dt>
+            <dd>
+              The vendor ID to match.
+            </dd>
+            <dt>
+              <dfn>productId</dfn> member
+            </dt>
+            <dd>
+              The product ID to match.
+            </dd>
+            <dt>
+              <dfn>usagePage</dfn> member
+            </dt>
+            <dd>
+              The [=top-level collection=] [=Usage Page=] to match.
+            </dd>
+            <dt>
+              <dfn>usage</dfn> member
+            </dt>
+            <dd>
+              The [=top-level collection=] [=Usage ID=] to match.
+            </dd>
+          </dl>
+          <p>
+            A {{HIDDevice}} |device:HIDDevice| <dfn data-lt=
+            "match the filter">matches the filter</dfn>
+            |filter:HIDDeviceFilter| if these steps return `true`:
+          </p>
+          <ol>
+            <li>If |filter|["{{HIDDeviceFilter/vendorId}}"] is present:
+              <ol>
+                <li>If |filter| contains "{{HIDDeviceFilter/productId}}" and
+                |device|["{{HIDDevice/productId}}"] is not equal to
```

Comment:
A HID device might not have a vendor or product ID.

Reply:
If the device has no vendor or product ID then the values for vendorId and productId should be zero.